Signal/src/textsecure/Messages/TSMessagesManager.m

```diff
@@ -179,6 +179,21 @@ - (void)handlePreKeyBundle:(IncomingPushMessageSignal*)preKeyMessage{
 }
 
 - (void)handleIncomingMessage:(IncomingPushMessageSignal*)incomingMessage withPushContent:(PushMessageContent*)content{
+    if(content.group!= nil )  {
+        __block BOOL ignoreMessage = NO;
+        [self.dbConnection readWithBlock:^(YapDatabaseReadTransaction *transaction) {
+            GroupModel *emptyModelToFillOutId = [[GroupModel alloc] initWithTitle:nil memberIds:nil image:nil groupId:content.group.id]; // TODO refactor the TSGroupThread to just take in an ID (as it is all that it uses). Should not take in more than it uses
+            TSGroupThread *gThread = [TSGroupThread threadWithGroupModel:emptyModelToFillOutId transaction:transaction];
+            if(gThread==nil) {
+                ignoreMessage = YES;
+            }
+        }];
+        if(ignoreMessage) {
+            DDLogDebug(@"recevied message from group that I left, ignoring");
```

Comment:
I think we wil lwant to carefully document this handling, though it is out
of protocol it is important (also for Masha!) that all of our clients have
the same behavior/edge cases, otherwise support and debugging will be a
pain and security properties may be different.

On Fri, Jan 2, 2015 at 9:25 PM, Christine Corbett Moran <
christine.corbett@gmail.com> wrote:

> cool! yea I would like to hear from Moxie his thoughts. in the meantime I
> will put out an example PR, I will be finished in a few minutes so that we
> can see what it would look like in this formulation.
>
> On Fri, Jan 2, 2015 at 9:22 PM, Frederic Jacobs <notifications@github.com>
> wrote:
>
>> In Signal/src/textsecure/Messages/TSMessagesManager.m
>> <https://github.com/WhisperSystems/Signal-iOS/pull/248#discussion-diff-22423363>
>> :
>>
>> > @@ -179,6 +179,21 @@ - (void)handlePreKeyBundle:(IncomingPushMessageSignal*)preKeyMessage{
>> >  }
>> >
>> >  - (void)handleIncomingMessage:(IncomingPushMessageSignal*)incomingMessage withPushContent:(PushMessageContent*)content{
>> > +    if(content.group!= nil )  {
>> > +        __block BOOL ignoreMessage = NO;
>> > +        [self.dbConnection readWithBlock:^(YapDatabaseReadTransaction *transaction) {
>> > +            GroupModel *emptyModelToFillOutId = [[GroupModel alloc] initWithTitle:nil memberIds:nil image:nil groupId:content.group.id]; // TODO refactor the TSGroupThread to just take in an ID (as it is all that it uses). Should not take in more than it uses
>> > +            TSGroupThread *gThread = [TSGroupThread threadWithGroupModel:emptyModelToFillOutId transaction:transaction];
>> > +            if(gThread==nil) {
>> > +                ignoreMessage = YES;
>> > +            }
>> > +        }];
>> > +        if(ignoreMessage) {
>> > +            DDLogDebug(@"recevied message from group that I left, ignoring");
>>
>> Log is redacted and contained only relevant points to this discussion but
>> I get your point.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/WhisperSystems/Signal-iOS/pull/248/files#r22423363>.
>>
>
>
>
> --
> Dr. Corbett Moran
> Rockets @ http://www.spacex.com
> Physics @ http://www.ICS.uzh.ch  <http://www.ICS.uzh.ch>
> Dev @ http://circleof6app.com
> Dev @ https://whispersystems.org
> www.christinecorbettmoran.com
>
>


-- 
Dr. Corbett Moran
Rockets @ http://www.spacex.com
Physics @ http://www.ICS.uzh.ch  <http://www.ICS.uzh.ch>
Dev @ http://circleof6app.com
Dev @ https://whispersystems.org
www.christinecorbettmoran.com